style.md

```diff
@@ -965,6 +966,83 @@ func (f *foo) isRunning() bool {
 </td></tr>
 </tbody></table>
 
+### Avoid Mutable Globals
+
+Avoid mutating global variables, instead opting for dependency injection.
+This applies to function pointers as well as other kinds of values.
+
+<table>
+<thead><tr><th>Bad</th><th>Good</th></tr></thead>
+<tbody>
+<tr><td>
+
+```go
+// foo.go
+
+var _doFoo = foo
```

Comment:
can we make this example a little more realistic, maybe a `timeNow func() time.Duration` function, since that's commonly stubbed in tests

Reply:
Going to switch to time.Now for message signing.

Reply:
Thanks @abhinav  -- should we use dependency injection here rather than changing the unexported field directly? That would also go better with the recommendation.

Reply:
I think @nomis52 here was specifically going for the "set the field" case.
Proper dependency injection is preferable but for this example, we would then
have to take a position on a clock library.  Right now, the example takes the
form of a privately injected dependency.  That isn't perfect but it's better
than the global form.

I can make it a parameter of the function (whether as a func or an interface),
but I don't know if that's the best idea here since we probably wouldn't do
that in practice depending on how coupled time is to the contract of the
abstraction.

Reply:
I've realized we need a separate section on dependency injection. My position is that for unexported types it's ok to set fields directly.

For exported types we should be using the Options pattern.

Thoughts?

Reply:
That's reasonable. The options pattern does allow for private options that are available to tests only. In the interest of keeping this example simple, we can make the object private.

Reply:
Fair enough, it's simpler to set fields, so let's go with that, and separately explore recommendations for dependency injection.